doc/howtowritearecogmethod.autodoc

```diff
@@ -0,0 +1,181 @@
+@Chapter How to write a recognition method
+
+This chapter explains how to integrate a newly developed group recognition
+method into the framework provided by the recog package.
+
+TODO:
+Refer to Chapter <Ref Chap="methsel"/> for an explanation of methods.
+There are leaf methods and split methods.
+The next two sections describe how to implement leaf and split methods
+respectively, and include example code.
+
+@Section Leaf methods
+
+A leaf method must at the very least do the following:
+
+- Provide the order of the recognized group via `SetSize(ri, NNN)`.
+- Provide a set of SLPs which map the original generators $X$ to the nice generators $Y$, as entry for the attribute `slptonice`.
+- Provide a function which maps any element $g\in G$ to a corresponding SLP in terms of the nice generators $Y$, as entry for the attribute `slpforelement`.
+- Call `SetFilterObj(ri, IsLeaf);` to mark the node as a leaf node.
+
+There are further values that can be provided, in particular to speed up
+computations; we'll come back to that later. Let's first look at an example:
+The following method is used by <Package>recog</Package> to recognize trivial
+groups, as a base case for the recursive group recognition algorithm. It works
+for arbitrary groups.
+
+TODO: AutoDoc inserts extra paragraph commands here:
+@InsertCode FindHomMethodsGeneric.TrivialGroup
+
+The input is in the format described above (TODO), and the return value is "Success".
+
+Two more comments:
+- When we check whether all generators are the identity, we call `ri!.isone`,
```

Comment:
We also use `isone(ri)`. I do not know which version is preferred.

Reply:
Both are used in recog code right now.

Formally, since `isone` is an attribute, the "official" way to access it is `isone(ri)`, while using `ri!.isone` relies on an implementation detail. HOWEVER, I personally find `ri!.isone(x)` much clearer than `isone(ri)(x)`, and even more so for `ri!.isequal(x,y)` vers. `isequal(ri)(x,y)`.

That said, it doesn't really matter much... My hope is that on the long run, we can get rid of these again, by using "proper" projective matrices (see #82); and before that, perhaps they get renamed (see #97).